### PR TITLE
Clean up unused CRL entries when issuer is removed

### PR DESCRIPTION
**Pull request by cipherboy**
When a issuer is removed, the space utilized by its CRL was not freed, both from the CRL config mapping issuer IDs to CRL IDs and from the CRL storage entry. We thus implement a two step cleanup, wherein orphaned CRL IDs are removed from the config and any remaining full CRL entries are removed from disk.

This relates to a Consul<->Vault interop issue (#22980), wherein Consul creates a new issuer on every leadership election, causing this config to grow. Deleting issuers manually does not entirely solve this problem as the config does not fully reclaim space used in this entry.

Notably, an observation that when deleting issuers, the CRL was rebuilt on secondary clusters (due to the invalidation not caring about type of the operation); for consistency and to clean up the unified CRLs, we also need to run the rebuild on the active primary cluster that deleted the issuer as well.

This approach does allow cleanup on existing impacted clusters by simply rebuilding the CRL.

---

TODO:

 - [x] Tests

### CODE REVIEW

File: builtin/logical/pki/crl_test.go [hunk 1]
```diff
@@ -1436,3 +1439,87 @@ hbiiPARizZA/Tsna/9ox1qDT
...
+	}
+	for _, key := range keyIDs {
+		_, err := CBDelete(b, s, "key/"+key)
+		require.NoError(t, err)
+	}
+
+	// Finally list storage entries again to ensure they are cleaned up.
+	afterCRLList, err := s.List(ctx, "crls/")
+	require.NoError(t, err)
+	for _, entry := range crlList {
```
Comment (stevendpclark) [inline builtin/logical/pki/crl_test.go:1516]: Should we add a test to validate that size of `afterCRLList` and `crlList` are equal? Right now I believe this only verifies that we didn't delete the original issuer's CRL?

Comment (cipherboy) [reply builtin/logical/pki/crl_test.go:1516]: Ah, that's probably good.

**Review by stevendpclark — APPROVED** [review]
👍

**Review by kitography — APPROVED** [review]
LGTM

**Pull request merged**